packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -333,10 +342,21 @@ Future<String> _initializeGradle(FlutterProject project) async {
     );
   } on ProcessException catch (e) {
     final String error = e.toString();
-    if (error.contains('java.io.FileNotFoundException: https://downloads.gradle.org') ||
-        error.contains('java.io.IOException: Unable to tunnel through proxy')) {
+    if (_kKnownErrorPrefixes.any((String candidate) => error.contains(candidate))) {
       throwToolExit('$gradle threw an error while trying to update itself.\n$e');
```

Comment:
Maybe advise to just try again. That's fixed this for me in the past. We might also add a TODO here to just do the retry automatically.

Reply:
Done